astropy/modeling/tests/test_functional_models_quantities.py

```diff
@@ -0,0 +1,244 @@
+        pytest.skip()
+    m = model['class'](**model['parameters'])
+    for args in model['evaluation']:
+        assert_quantity_allclose(m(*args[:-1]), args[-1])
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_with_units_x_array(model):
+
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+
+    m = model['class'](**model['parameters'])
+
+    for args in model['evaluation']:
+        if len(args) == 2:
+            x, y = args
+            x_arr = u.Quantity([x, x])
+            result = m(x_arr)
+            assert_quantity_allclose(result, u.Quantity([y, y]))
+        else:
+            x, y, z = args
+            x_arr = u.Quantity([x, x])
+            y_arr = u.Quantity([y, y])
+            result = m(x_arr, y_arr)
+            assert_quantity_allclose(result, u.Quantity([z, z]))
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_evaluatate_with_units_param_array(model):
+
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+
+    params = {}
+    for key, value in model['parameters'].items():
+        if value is None:
+            params[key] = None
+        else:
+            params[key] = np.repeat(value, 2)
+
+    m = model['class'](**params)
+    for args in model['evaluation']:
+        if len(args) == 2:
+            x, y = args
+            x_arr = u.Quantity([x, x])
+            result = m(x_arr)
+            assert_quantity_allclose(result, u.Quantity([y, y]))
+        else:
+            x, y, z = args
+            x_arr = u.Quantity([x, x])
+            y_arr = u.Quantity([y, y])
+            result = m(x_arr, y_arr)
+            assert_quantity_allclose(result, u.Quantity([z, z]))
+
+
+@pytest.mark.parametrize('model', MODELS)
+def test_models_bounding_box(model):
+
+    if not HAS_SCIPY and model['class'] in SCIPY_MODELS:
+        pytest.skip()
+
+    m = model['class'](**model['parameters'])
+
+    if model['bounding_box']:
+        m.bounding_box
+    else:
+        # Check that NotImplementedError is raised, so that if bounding_box is
+        # implemented we remember to set bounding_box=True in the list of models
+        # above
+        with pytest.raises(NotImplementedError):
+            m.bounding_box
+
+
+@pytest.mark.skipif('not HAS_SCIPY')
+@pytest.mark.parametrize('model', MODELS)
+def test_models_fitting(model):
+    m = model['class'](**model['parameters'])
+    if len(model['evaluation'][0]) == 2:
+        x = np.linspace(1, 3, 100) * model['evaluation'][0][0].unit
+        y = np.exp(-x.value ** 2) * model['evaluation'][0][1].unit
+        args = [x, y]
+    else:
+        x = np.linspace(1, 3, 100) * model['evaluation'][0][0].unit
+        y = np.linspace(1, 3, 100) * model['evaluation'][0][1].unit
+        z = np.exp(-x.value**2 - y.value**2) * model['evaluation'][0][2].unit
+        args = [x, y, z]
+    fitter = LevMarLSQFitter()
+    m_new = fitter(m, *args)
```

Comment:
That is correct - I should at least make sure that the resulting models have units - however I think checking the fitted parameter values is beyond the scopes of these tests